src/main/scala/org/broadinstitute/dsde/firecloud/core/ProfileClient.scala

```diff
@@ -84,77 +82,101 @@ class ProfileClientActor(requestContext: RequestContext) extends Actor with Fire
-                      // TODO: we don't really need to check both lastLinkTime and linkExpireTime here.
-                        // but, until the main OAuth login sets linkExpireTime, we need it for safety.
-                      case Some(x) =>
-                        (profile.lastLinkTime, profile.linkExpireTime) match {
-                          case (Some(lastLinkSeconds:Long), Some(linkExpireSeconds:Long)) =>
-                            val howOld = DateUtils.hoursSince(lastLinkSeconds)
-                            val howSoonExpire = DateUtils.secondsSince(linkExpireSeconds)
-
-                            val loginRequired = (howOld >= 24 || howSoonExpire >= 0)
-
-                            val secsSinceLastLink = DateUtils.secondsSince(lastLinkSeconds)
-
-                            val descSince = DateUtils.prettySince(lastLinkSeconds)
-
-                            val statusResponse = NIHStatus(
-                              loginRequired,
-                              linkedNihUsername = profile.linkedNihUsername,
-                              isDbgapAuthorized = profile.isDbgapAuthorized,
-                              lastLinkTime = Some(lastLinkSeconds),
-                              linkExpireTime = profile.linkExpireTime,
-                              secondsSinceLastLink = Some(secsSinceLastLink),
-                              descriptionSinceLastLink = Some(descSince))
-
-                            RequestComplete(StatusCodes.OK, statusResponse)
-                          case _ =>
-                            // user is linked and authorized, but we have no record of login time or expiration time.
-                            RequestComplete(StatusCodes.OK, NIHStatus(true,
-                              linkedNihUsername = profile.linkedNihUsername,
-                              lastLinkTime = profile.lastLinkTime,
-                              linkExpireTime = profile.linkExpireTime,
-                              isDbgapAuthorized = profile.isDbgapAuthorized))
-
-                        }
-                      case _ => RequestComplete(NoContent, "Not dbGaP authorized")
+    case GetAndUpdateNIHStatus(userInfo: UserInfo) =>
+      getProfile(userInfo, true)
+      // do NOT pipe the response back to the parent!
+  }
+
+  def getProfile(userInfo: UserInfo, updateExpiration: Boolean): Future[PerRequestMessage] = {
+    val pipeline = addCredentials(userInfo.accessToken) ~> sendReceive
+    val profileReq = Get(UserService.remoteGetAllURL.format(userInfo.getUniqueId))
+
+    pipeline(profileReq) map { response: HttpResponse =>
+      response.status match {
+        case OK =>
+          val profileEither = response.entity.as[ProfileWrapper]
+          profileEither match {
+            case Right(profileWrapper) =>
+              val profile = Profile(profileWrapper)
+              profile.linkedNihUsername match {
+                case Some(nihUsername) =>
+                  // we have a linked profile.
+
```

Comment:
most of the getProfile function is a cut-and-paste from the previous GetNIHStatus case. However, note I've removed the check on profile.isDbgapAuthorized, which added very little. See DSDEEPB-2404 for more context.